dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/AbstractEventService.java

```diff
@@ -833,28 +835,53 @@ public ImportSummaries deleteEvents( List<String> uids, boolean clearSession )
     // HELPERS
     // -------------------------------------------------------------------------
 
-    private List<OrganisationUnit> getOrganisationUnits( EventSearchParams params )
+    private List<OrganisationUnit> getOrganisationUnits( EventSearchParams params, User user )
     {
         List<OrganisationUnit> organisationUnits = new ArrayList<>();
 
+        Program program = params.getProgram();
         OrganisationUnit orgUnit = params.getOrgUnit();
         OrganisationUnitSelectionMode orgUnitSelectionMode = params.getOrgUnitSelectionMode();
 
         if ( params.getOrgUnit() != null )
         {
+            organisationUnits.add( orgUnit );
+
             if ( OrganisationUnitSelectionMode.DESCENDANTS.equals( orgUnitSelectionMode ) )
             {
                 organisationUnits.addAll( organisationUnitService.getOrganisationUnitWithChildren( orgUnit.getUid() ) );
             }
             else if ( OrganisationUnitSelectionMode.CHILDREN.equals( orgUnitSelectionMode ) )
             {
-                organisationUnits.add( orgUnit );
                 organisationUnits.addAll( orgUnit.getChildren() );
             }
-            else // SELECTED
+        }
+
+        if ( user != null )
+        {
+            Set<OrganisationUnit> orgUnits = new HashSet<>();
+
+            if ( OrganisationUnitSelectionMode.ACCESSIBLE.equals( orgUnitSelectionMode ) )
+            {
+                orgUnits = user.getTeiSearchOrganisationUnitsWithFallback();
```

Comment:
I think lines 866 and 870 are mutually exclusive so it would be okay if we put them in if-else.
Other than this, all seems good to me.

Reply:
The plan is we go with tei search scope if we have no information about program. This also goes well if program happens to be with registration. However, if program exists and without registration then we go with data view.